lib/browser.ts

```diff
@@ -193,24 +193,18 @@ export class ProtractorBrowser extends AbstractExtendedWebDriver {
    * this method is called use the new app root. Pass nothing to get a promise that
    * resolves to the value of the selector.
    *
-   * @param {string|webdriver.promise.Promise<string>} value The new selector.
+   * @param {string|webdriver.promise.Promise<string>} valuePromise The new selector.
    * @returns A promise that resolves with the value of the selector.
    */
-  angularAppRoot(value: string|wdpromise.Promise<string> = null): wdpromise.Promise<string> {
-    return this.driver.controlFlow().execute(() => {
-      if (value != null) {
-        return wdpromise.when(value).then((value: string) => {
-          this.internalRootEl = value;
-          if (this.bpClient) {
-            const bpCommandPromise = this.bpClient.setWaitParams(value);
-            // Convert to webdriver promise as best as possible
-            return wdpromise.when(bpCommandPromise as any).then(() => this.internalRootEl);
-          }
-          return this.internalRootEl;
-        });
+  async angularAppRoot(valuePromise: string|wdpromise.Promise<string> = null): Promise<string> {
+    if (valuePromise != null) {
+      const value = await Promise.resolve(valuePromise);
```

Comment:
You don't need the Promise.resolve. You can just `await valuePromise`. If It's not a promise, await will just return the string.

Reply:
Neat!